app/src/ValidateCLIParameter.cpp

```diff
@@ -0,0 +1,193 @@
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+------------------------------------------------------------------------------*/
+
+#include "OpenConfiguratorCore.h"
+#include "ValidateCLIParameter.h"
+#include "CLIResult.h"
+#include "LoggingConfiguration.h"
+#include "Parser.h"
+
+using namespace IndustrialNetwork::POWERLINK::Core::API;
+using namespace IndustrialNetwork::POWERLINK::openCONFIGURATORCLI::ErrorHandling;
+using namespace IndustrialNetwork::POWERLINK::Core::CoreConfiguration;
+
+
+boost::format fileParam(kMsgProjectFileExists);
+boost::format toolformatter(kMsgToolGenerator);
+boost::format vendorformatter(kMsgToolVendor);
+boost::format usage(kMsgToolUsage);
+boost::format options(kMsgOptions);
+boost::format projectParam(kMsgProjectParameter);
+boost::format outputParam(kMsgOutputparameter);
+boost::format langParam(kMsgLanguageparameter);
+boost::format logParam(kMsgLogMessage);
+boost::format helpParam(kMsgHelpParameter);
+boost::format insufficientParam(kMsgInsufficientParameters);
+boost::format invalidFileParam(kMsgInvalidFile);
+boost::format outputDirFileParam(kMsgOutputDirectoryExists);
+boost::format invalidParam(kMsgInvalidparameter);
+
+
+void ValidateCLIParameter::Showusage()
+{
+	LOG_INFO() << toolformatter.str();
+	LOG_INFO() << vendorformatter.str();
+	LOG_INFO() << usage.str();
+	LOG_INFO() << options.str();
+	LOG_INFO() << projectParam.str();
+	LOG_INFO() << outputParam.str();
+	LOG_INFO() << langParam.str();
+	LOG_INFO() << logParam.str();
+	LOG_INFO() << helpParam.str();
+	exit(1);
+}
+
+ValidateCLIParameter& ValidateCLIParameter::GetInstance()
+{
+	static ValidateCLIParameter instance;
+	return instance;
+}
+
+std::string ValidateCLIParameter::GetProjectpath()
+{
+	return std::string(ValidateCLIParameter::xmlFile);
+}
+
+std::string ValidateCLIParameter::GetNetworkName()
+{
+	return boost::filesystem::basename(ValidateCLIParameter::xmlFile);
+}
+
+int ValidateCLIParameter::ValidateParameter(int parameterCount, char* parameter[])
+{
+	std::string projectXMLFile = "";
+
+	if (parameterCount < 4)
+	{
+		boost::format formatter(kMsgInvalidArguments);
+		LOG_ERROR() << "Invalid number of arguments. Expected: 3 Actual: " << parameterCount;
+		Showusage();
+		exit(1);
+	}
+	else if (parameterCount == 1)
+	{
+		Showusage();
+		exit(1);
+	}
+	else if (parameterCount == 2)
+	{
+		LOG_ERROR() << insufficientParam.str();
+		Showusage();
+		exit(1);
+	}
+
+	std::string projectFileparam = std::string(parameter[1]);
+
+	if ((projectFileparam == "-p") || (projectFileparam == "--project"))
+	{
+
+		projectXMLFile = std::string(parameter[2]);
+		ValidateCLIParameter::xmlFile = parameter[2];
+		if (!boost::filesystem::exists(projectXMLFile))
+		{
+			LOG_ERROR() << fileParam.str();
+			exit(1);
+		}
+
+		if (boost::filesystem::extension(projectXMLFile) != ".xml")
+		{
+
+			LOG_ERROR() << invalidFileParam.str();
+			exit(1);
+		}
+		if (parameter[3] != NULL)
```

Comment:
All NULL checks are useless because the number of arguments shows the presence of non null entries in the argument array. The arguments can be empty.